### PR TITLE
ci: add remaining steps in publish workflow TDE-716

### PR DESCRIPTION
**Pull request by paulfouquet**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -6,20 +6,57 @@ jobs:
...
+        run: git diff --exit-code
+
+      # TODO: Remove and modify "Create Catalog" once argo-tasks released
+      - name: Get Collections
+        run: |
+          echo "$(find stac/ -name collection.json)" > files.txt
+
+      # catalog.json is not pushed to the repository (temporary solution)
+      - name: Create Catalog
+        uses: docker://ghcr.io/linz/argo-tasks:v2
```
Comment (blacha) [inline .github/workflows/publish.yml:38]: I thought t he latest version of this just took the path to the stac folder? or has it not been merged yet?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -6,20 +6,57 @@ jobs:
...
+        with:
+          args: stac-catalog --output stac/catalog.json --template template/catalog.json --collections files.txt
+
+      - name: Validate STAC
+        run: |
+          # Enable double star operator
+          shopt -s globstar
+          docker run -v $PWD:$PWD ghcr.io/linz/argo-tasks:v2 stac-validate $PWD/stac/**/collection.json
+
+      # FIXME: issue with the latest version (v2.0.0): https://github.com/aws-actions/configure-aws-credentials/issues/680
```
Comment (blacha) [inline .github/workflows/publish.yml:48]: That is a deprecation warning? What affect does it have on using t he master version?

File: .github/workflows/publish.yml (hunk 1 above)
Comment (paulfouquet) [reply .github/workflows/publish.yml:38]: It has been merged. I'll update that

File: .github/workflows/publish.yml (hunk 2 above)
Comment (paulfouquet) [reply .github/workflows/publish.yml:48]: Weirdly I had an issue while using `v2` which was actually related to the `permission` that I haven't set. So not related to that at all... I'll upgrade.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (paulfouquet) [reply .github/workflows/publish.yml:38]: `stac-catalog --output stac/catalog.json --template template/catalog.json stac/` errors with 
```
Error: FilePaths are not relative base: stac/ file: /github/workspace/stac/auckland/auckland_2010-2012_0.5m/rgb/2193/collection.json
```
I passed the full absolute path to workaround it
`stac-catalog --output stac/catalog.json --template template/catalog.json /github/workspace/stac/`

**Review by blacha — APPROVED** [review]

**Pull request merged**